sycl/include/CL/sycl/detail/platform_impl.hpp

```diff
@@ -24,88 +23,101 @@ class device;
+// TODO: implement parameters treatment for host device
 class platform_impl {
 public:
-  platform_impl() = default;
+  /// Constructs platform_impl for a SYCL host platform.
+  platform_impl() : MHostPlatform(true) {}
 
-  explicit platform_impl(const device_selector &);
+  /// Constructs platform_impl from a plug-in interoperability platform handle.
+  ///
+  /// @param Platform is a raw plug-in platform handle.
+  explicit platform_impl(RT::PiPlatform Platform) : MPlatform(Platform) {}
 
-  virtual bool has_extension(const string_class &extension_name) const = 0;
+  ~platform_impl() = default;
 
-  virtual vector_class<device>
-      get_devices(info::device_type = info::device_type::all) const = 0;
+  /// Checks if this platform supports extension.
+  ///
+  /// @param ExtensionName is a string containing extension name.
+  /// @return true if platform supports specified extension.
+  bool has_extension(const string_class &ExtensionName) const {
+    if (is_host())
+      return false;
 
+    string_class all_extension_names =
+        get_platform_info<string_class, info::platform::extensions>::get(
+            MPlatform);
+    return (all_extension_names.find(ExtensionName) != std::string::npos);
+  }
+
+  /// Returns all SYCL devices associated with this platform.
+  ///
+  /// If this platform is a host platform, resulting vector contains only
+  /// a single SYCL host device. If there are no devices that match given device
+  /// type, resulting vector is empty.
+  ///
+  /// @param DeviceType is a SYCL device type.
+  /// @return a vector of SYCL devices.
+  vector_class<device>
+  get_devices(info::device_type DeviceType = info::device_type::all) const;
+
+  /// Queries this SYCL platform for info.
+  ///
+  /// The return type depends on information being queried.
   template <info::platform param>
   typename info::param_traits<info::platform, param>::return_type
   get_info() const {
-    if (is_host()) {
+    if (is_host())
       return get_platform_info_host<param>();
-    }
+
     return get_platform_info<
         typename info::param_traits<info::platform, param>::return_type,
         param>::get(this->getHandleRef());
   }
 
-  virtual bool is_host() const = 0;
-
-  virtual cl_platform_id get() const = 0;
-
-  // Returns underlying native platform object.
-  virtual const RT::PiPlatform &getHandleRef() const = 0;
+  /// Checks if this platform is a host platform.
+  ///
+  /// @return true if this SYCL platform is a host platform.
+  bool is_host() const { return MHostPlatform; };
 
-  virtual ~platform_impl() = default;
-};
-
-// TODO: merge platform_impl_pi, platform_impl_host and platform_impl?
-class platform_impl_pi : public platform_impl {
-public:
-  platform_impl_pi(RT::PiPlatform a_platform) : m_platform(a_platform) {}
-
-  vector_class<device> get_devices(
-      info::device_type deviceType = info::device_type::all) const override;
+  /// Returns an OpenCL interoperability platform.
+  ///
```

Comment:
I think Doxygen may emit warnings otherwise

Reply:
I don't think that it is a valid justification to add a lot of boilerplate documentation. It doesn't provide any new information.

Reply:
Agree with Andrew, it would be good if we can configure Doxygen to allow empty brief description, or, we can just omit `@return` directive.

Also I believe that some of methods are self-explanatory